File: src/primus/pawel/Exercise_162.java
package primus.pawel;

import java.util.LinkedList;

// What's in a name
// 6kyu
// https://www.codewars.com/kata/59daf400beec9780a9000045/train/java
// 2021-06-24 T:22:21:40
class Exercise_162 {

    public static boolean nameInStr(String str, String name) {
        LinkedList<Character> charsQue = new LinkedList();
        for (int i = 0; i < name.length(); i++) {
            charsQue.add(name.charAt(i));
        }

        int i = 0;
        while (i < str.length() && charsQue.size() > 0) {
            if (str.charAt(i) == charsQue.getFirst()) {
                charsQue.remove();
            }
            i++;
        }
        return (charsQue.size() == 0);
    }


    public static void main(String[] args) {
        System.out.println(nameInStr("cNext hto a lake", "chris"));

    }
}

File: src/JS/Exercise_162.js
// Product of Array Items
// 7kyu
// https://www.codewars.com/kata/5901f361927288d961000013/train/javascript
// 2022-07-07 T:07:43:35

function product(values) {
    if(values == null || values.length === 0 ){
        return null;
    }
    return values.reduce((a,b) => a*b , 1);
}

console.log(product([5, 4, 1, 3, 9]));